resolver/resolver.go

```diff
@@ -107,54 +106,49 @@ func (res *Resolver) Serve(proto string) (<-chan struct{}, <-chan error) {
 	return ch, errCh
 }
 
-// LaunchZK launches a Zookeeper detector, returns immediately two chans: the first fires an empty
-// struct whenever there's a new (non-nil) mesos leader, the second if there's an unrecoverable
-// error in the master detector.
-func (res *Resolver) LaunchZK(initialDetectionTimeout time.Duration) (<-chan struct{}, <-chan error) {
-	var startedOnce sync.Once
-	startedCh := make(chan struct{})
-	errCh := make(chan error, 1)
-	leaderCh := make(chan struct{}, 1) // the first write never blocks
+// LaunchZK launches a Zookeeper detector and returns immediately two channels,
+// the first fires master changes,
+// the second if there's an unrecoverable error in the detector.
+func (res *Resolver) LaunchZK(initialDetectionTimeout time.Duration) (<-chan []string, <-chan error) {
```

Comment:
I don't really see the need for this whole method which is just wrapping a channel receive with a timeout which we could simply do in `main`.